File: src/api/schemas/product.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ProductBase(BaseModel):
    name: str
    description: Optional[str] = None
    category: str
    tags: Optional[str] = None
    price: float
    locked: bool = True
    popular: bool = False
    preview_available: bool = True


class ProductResponse(ProductBase):
    id: int
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

    class Config:
        orm_mode = True

class ProductCreate(ProductBase):
    pass

File: src/web/src/models/Product.ts
export type Product = {
	id: number;
	name: string;
	description: string | null;
	category: string;
	tags: string | null;
	price: number;
	locked: boolean;
	popular: boolean;
	preview_available: boolean;
	created_at: string | null;
	updated_at: string | null;
};
